core/domain/user_services.py

```diff
@@ -920,6 +940,63 @@ def has_fully_registered_account(user_id):
         feconf.REGISTRATION_PAGE_LAST_UPDATED_UTC)
 
 
+def get_all_profiles_auth_details_by_parent_user_id(parent_user_id):
+    """Gets domain objects representing the auth details for all profiles
+    associated with the user having given parent_user_id.
+
+    Args:
+        parent_user_id: str. User id of the parent_user whose associated
+            profiles we are querying for.
+
+    Returns:
+        list(UserAuthDetails). The UserAuthDetails domain objects
+        corresponding to the profiles linked to given parent_user_id. If that
+        parent user does not have any profiles linked to it, the
+        returned list will be empty.
+    """
+    return [
+        _get_user_auth_details_from_model(model) for model in
+        user_models.UserAuthDetailsModel.get_all_profiles_by_parent_user_id(
+            parent_user_id)
+    ]
+
+
+def is_display_alias_unique_within_the_account(parent_user_id, display_alias):
+    """Checks whether a given display alias is unique within the account.
```

Comment:
I suggest being more specific: this is checking for uniqueness among all profiles associated with the specified parent user ID, including that user. 'within the account' is a bit vague, and arguably incorrect since each profile technically is its own user.

Reply:
Updated the function name to `does_profile_with_display_alias_exist`.
Yes I agree each profile is technically its own user but they are not an account right? I hope the current name and documentation makes sense.